# 1. Площадь треугольника
Напишите программу, которая считывает длины двух катетов в прямоугольном треугольнике и выводит его площадь.
![](https://ucarecdn.com/c2fcd9b1-7105-4a8f-af91-0ede0f9f5b2e/)

`Формат входных данных`

На вход программе подаётся два числа с плавающей точкой – длины катетов, каждое на отдельной строке.

`Формат выходных данных`

Программа должна вывести одно число – площадь треугольника.

In [1]:
a, b = float(input()), float(input())
s = 0.5 * a * b
print(s)

32.2
25.5
410.55


# 2. Две старушки
Две старушки идут навстречу друг другу с постоянными скоростями V1 и V2 км/ч. Определите, через какое время старушки встретятся, если расстояние между ними равно S км.
![](https://ucarecdn.com/ac06613a-0c35-4020-9116-323b77f37494/)

`Формат входных данных`

На вход программе подаются три числа с плавающей точкой S, V1 и V2, каждое на отдельной строке.

`Формат выходных данных `

Программа должна вывести одно число в соответствии с условием задачи.

In [3]:
s, v1, v2 = float(input()), float(input()), float(input())
t = s / (v1 + v2)
print(t)

60.0
15.5
14.5
2.0


# 3. Обратное число
Напишите программу, которая считывает с клавиатуры одно число и выводит обратное ему. Если при этом введённое с клавиатуры число – ноль, то вывести «Обратного числа не существует» (без кавычек).
![](https://ucarecdn.com/4df00cbb-37df-4955-9f01-4da08fd44cad/)

`Формат входных данных`

На вход программе подается одно действительное число.

`Формат выходных данных`

Программа должна вывести действительное число обратное данному, либо текст в соответствии с условием задачи.

In [4]:
x = float(input())
if x == 0:
    print("Обратного числа не существует")
elif x != 0:
    print(1 / x)

4
0.25


# 4. 451 градус по Фаренгейту 
У известного американского писателя Рэя Бредбери есть роман «451 градус по Фаренгейту». Напишите программу, которая определяет, какой температуре по шкале Цельсия соответствует указанное значение по шкале Фаренгейта.

Используйте формулу для перевода: C = 5/9 * (F - 32)

`Формат входных данных`

На вход программе подаётся вещественное число градусов по шкале Фаренгейта.

`Формат выходных данных`

Программа должна вывести число градусов по шкале Цельсия.

In [5]:
x = float(input())
c = 5 / 9 * (x - 32)
print(c)

113.75
45.41666666666667


# 5. Dog age
На вход программе подается число nn – количество собачьих лет. Напишите программу, которая вычисляет возраст собаки в человеческих годах.

`Формат входных данных`

На вход программе подаётся натуральное число – количество собачьих лет.

`Формат выходных данных`

Программа должна вывести возраст собаки в человеческих годах.

Примечание. В течение первых двух лет собачий год равен 10.5 человеческим годам. После этого каждый год собаки равен 4 человеческим годам.

In [6]:
x = float(input())
a = 10.5
b = 4
if x <= 2:
    print(a * x)
elif x > 2:
    print((a * 2) + (b * (x - 2)))

1
10.5


# 6. Первая цифра после точки
Дано положительное действительное число. Выведите его первую цифру после десятичной точки.

`Формат входных данных`

На вход программе подается положительное действительное число.

`Формат выходных данных`

Программа должна вывести цифру в соответствии с условием задачи.

In [7]:
x = float(input())
a = int(x * 10)
b = a % 10
print(b)

12.1322
1


# 7. Дробная часть
Дано положительное действительное число. Выведите его дробную часть.

`Формат входных данных`

На вход программе подается положительное действительное число.

`Формат выходных данных`

Программа должна вывести дробную часть числа в соответствии с условием задачи.

In [8]:
a = float(input())
b = a * 100
c = b % 100
d = c / 100
print(d)

44.45
0.45


# 8. Какое число будет выведено на экран в результате выполнения следующего кода?

num = max(1, 3, -5, 7) + min(-3, 6, -8, -1) + abs(-17)

print(num)

Ответ: `16`

# 9. Наибольшее и наименьшее
Напишите программу, которая находит наименьшее и наибольшее из пяти чисел.

`Формат входных данных`

На вход программе подается пять целых чисел, каждое на отдельной строке.

`Формат выходных данных`

Программа должна вывести наименьшее и наибольшее число с поясняющей надписью.

In [9]:
a, b, c, d, g = int(input()), int(input()), int(input()), int(input()), int(input())
e = min(a, b, c, d, g)
f = max(a, b, c, d, g)
print("Наименьшее число =", e)
print("Наибольшее число =", f)

33
213
5125
442
123
Наименьшее число = 33
Наибольшее число = 5125


# 10. Сортировка трёх 🌶️
Напишите программу, которая упорядочивает три числа от большего к меньшему.

`Формат входных данных`

На вход программе подается три целых числа, каждое на отдельной строке.

`Формат выходных данных`

Программа должна вывести три числа, каждое на отдельной строке, упорядоченных от большего к меньшему.

In [10]:
a, b, c = int(input()), int(input()), int(input())
d = min(a, b, c)
e = max(a, b, c)
f = a + b + c - (d + e)
print(e)
print(f)
print(d)

132
129
135
135
132
129


# 11. Интересное число
Назовем число интересным, если в нем разность максимальной и минимальной цифры равняется средней по величине цифре. Напишите программу, которая определяет интересное число или нет. Если число интересное, следует вывести – «Число интересное» иначе «Число неинтересное».

`Формат входных данных`

На вход программе подается целое трехзначное число.

`Формат выходных данных`

Программа должна вывести текст в соответствии с условием задачи.

In [11]:
a = int(input())
x = a // 100
y = a % 100 // 10
z = a % 100 % 10
b = min(x, y, z)
c = max(x, y, z)
d = x + y + z - (c + b)
if c - b == d:
    print("Число интересное")
else:
    print("Число неинтересное")

956
Число неинтересное


# 12. Абсолютная сумма
Даны пять чисел a1, a2, a3, a4, a5. Напишите программу, которая вычисляет сумму их модулей |a1| + |a2| +|a3| +|a4| + |a5|.

`Формат входных данных`

На вход программе подается пять действительных чисел a1, a2, a3, a4, a5, каждое на отдельной строке.

`Формат выходных данных` 

Программа должна вывести одно число – сумму модулей введённых чисел.

In [12]:
a, b, c, d, e = abs(float(input())), abs(float(input())), abs(float(input())), abs(float(input())), abs(float(input()))
g = a + b + c + d + e
print(g)

13
2.55
-125
-0.3123
444.1
584.9623


# 13. Манхэттенское расстояние
Прогуливаясь по Манхэттену, вы не можете попасть из точки А в точку Б по кратчайшему пути. Если только вы не умеете проходить сквозь стены, вам обязательно придется идти вдоль его параллельно-перпендикулярных улиц.

На плоскости манхэттенское расстояние между двумя точками (p1; p2) и (q1; q2) определяется так |p1 - q1| + |p2 - q2|.

Напишите программу определяющую манхэттенское расстояние между двумя точками, координаты которых заданы.
![](https://ucarecdn.com/d3720269-108b-48fb-9ca5-f24af59fb358/)

`Формат входных данных`

На вход программе подается четыре целых числа, каждое на отдельной строке – p1, p2, q1, q2

`Формат выходных данных`

Программа должна вывести одно число – манхэттенское расстояние.

In [13]:
a, b, c, d = int(input()), int(input()), int(input()), int(input())
e = abs(a - c) + abs(b - d)
print(e)

10
15
21
13
13


# 14. Что покажет приведенный ниже фрагмент кода?

mystr = 'да'

mystr = mystr + 'нет'

mystr = mystr + 'да'

print(mystr)

Ответ: `данетда`

# 15. Что покажет приведенный ниже фрагмент кода?

str1 = '1'

str2 = str1 + '2' + str1

str3 = str2 + '3' + str2

str4 = str3 + '4' + str3

print(str4)

Ответ: `121312141213121`


# 16. Что покажет приведенный ниже фрагмент кода?

mystr = '123' * 3 + '456' * 2 + '789' * 1

print(mystr)

Ответ: `123123123456456789`

# 17. Напишите программу, которая выводит текст:

"Python is a great language!", said Fred. "I don't ever remember having this much fun before."

Примечание. Используйте конкатенацию строк

In [14]:
text = '''"Python is a great language!", said Fred. "I don't ever remember having this much fun before."'''
print(text)

"Python is a great language!", said Fred. "I don't ever remember having this much fun before."


# 18. What's Your Name?
Напишите программу, которая считывает с клавиатуры две строки – имя и фамилию пользователя и выводит фразу:

«Hello [введенное имя] [введенная фамилия]! You just delved into Python».

`Формат входных данных`

На вход программе подаётся две строки (имя и фамилия), каждая на отдельной строке.

`Формат выходных данных`

Программа должна вывести текст в соответствии с условием задачи.

Примечание. Между firstname lastname вставьте пробел =)

In [15]:
fn, ln = input(), input()
print('Hello', fn, ln + '!', 'You just delved into Python')

Leonardo
DiCaprio
Hello Leonardo DiCaprio! You just delved into Python


# 19. Футбольная команда
Напишите программу, которая считывает с клавиатуры название футбольной команды и выводит фразу:

«Футбольная команда [введённая строка] имеет длину [длина введённой строки] символов».

`Формат входных данных`

На вход программе подаётся строка – название футбольной команды.

`Формат выходных данных`

Программа должна вывести текст в соответствии с условием задачи.



In [16]:
c = input()
l = len(c)
print("Футбольная команда", c, "имеет длину", l, "символов")

Barcelona
Футбольная команда Barcelona имеет длину 9 символов


# 20. Три города
Даны названия трех городов. Напишите программу, которая определяет самое короткое и самое длинное название города.

`Формат входных данных`

На вход программе подаётся названия трех городов, каждое на отдельной строке.

`Формат выходных данных`

Программа должна вывести самое короткое и длинное название города, каждое на отдельной строке.

Примечание. Гарантируется, что длины названий всех трех городов различны.



In [17]:
a, b, c = input(), input(), input()
x = len(a)
y = len(b)
z = len(c)
a1 = min(x, y, z)
a2 = max(x, y, z)
if y > x < z:
    print(a)
elif x > y < z:
    print(b)
elif x > z < y:
    print(c)
if y < x > z:
    print(a)
elif x < y > z:
    print(b)
elif x < z > y:
    print(c)

Париж
Марсель
Лион
Лион
Марсель


# 21. Арифметические строки
Вводятся 3 строки в случайном порядке. Напишите программу, которая выясняет можно ли из длин этих строк построить возрастающую арифметическую прогрессию.

`Формат входных данных`

На вход программе подаются три строки, каждая на отдельной строке.

`Формат выходных данных`

Программа должна вывести строку «YES», если из длин введенных слов можно построить арифметическую прогрессию, «NO» в ином

In [18]:
a, b, c = len(input()), len(input()), len(input())
if (2 * b - c - a) * (2 * c - b - a) * (2 * a - b - c) == 0:
    print("YES")
else:
    print("NO")

abc
a
abcde
YES


# 22. Какие значения может принимать строковая переменная s, чтобы в результате выполнения кода было выведено слово «YES»?

if s in 'abc123abc':

    print('YES')
    
else:

    print('NO')
    

* **s = 'a'**

* **s = '1'**

* s = 'abcabc'

* **s = '123abc'**

* **s = '3ab'**

* s = '321'

* **s = '23'**

* s = 'bca'

* s = 'bc2'

# 23. Цвет настроения синий
Напишите программу, которая считывает одну строку, после чего выводит «YES», если в введенной строке есть подстрока «синий» и «NO» в противном случае.

`Формат входных данных`

На вход программе подается одна строка.

`Формат выходных данных`

Программа должна вывести текст в соответствии с условием задачи.

In [19]:
a = input()
if "синий" in a:
    print("YES")
else:
    print("NO")

Какой хороший день!
NO


# 24. Отдыхаем ли?
Напишите программу, которая считывает одну строку, после чего выводит «YES», если в введённой строке есть подстрока «суббота» или «воскресенье», и «NO» в противном случае.

`Формат входных данных`

На вход программе подается одна строка.

`Формат выходных данных`

Программа должна вывести текст в соответствии с условием задачи.



In [20]:
a = input()
if "суббота" in a or "воскресенье" in a:
    print("YES")
else:
    print("NO")

Была суббота, и ему хотелось поскорее уехать домой.
YES


# 25. Корректный email
Будем считать email адрес корректным, если в нем есть символ собачки (@) и точки. Напишите программу проверяющую корректность email адреса.

`Формат входных данных`

На вход программе подаётся одна строка – email адрес.

`Формат выходных данных`

Программа должна вывести строку «YES», если email адрес является корректным и «NO» в ином случае.

Примечание. Наличие символов @ и . недостаточно для корректности email адреса, однако их отсутствие гарантировано влечёт за собой неверный email.

In [21]:
a = input()
if "@" in a and "." in a:
    print("YES")
else:
    print("NO")

aaaa@bbb.com
YES


# 26. Евклидово расстояние
На плоскости евклидово расстояние между двумя точками (x1; y1) и (x2; y2) определяется так p = sqrt(x1 - x2)^2 + (y1 - y2)^2.
![](https://ucarecdn.com/26c7ba17-bee9-4f8f-9764-3b68b2ede790/)
Напишите программу определяющую евклидово расстояние между двумя точками, координаты которых заданы.

`Формат входных данных`

На вход программе подается четыре вещественных числа, каждое на отдельной строке - x1, y1, x2, y2.

`Формат выходных данных`

Программа должна вывести одно число – евклидово расстояние.

In [22]:
from math import *
a, b, c, d = float(input()), float(input()), float(input()), float(input())
an = sqrt(pow(a - c, 2) + pow(b - d, 2))
print(an)

2.0
2.5
44.155
100.50
106.68197610187018


# 27. Площадь и длина
Напишите программу определяющую площадь круга и длину окружности по заданному радиусу R.

`Формат входных данных`

На вход программе подается одно вещественное число R.

`Формат выходных данных`

Программа должна вывести два числа – площадь круга и длину окружности радиуса R.
![](https://ucarecdn.com/4fa6a2d0-9522-4a29-8cac-b19e873f26c3/)
Примечание. Используйте константу math.pi. 

In [23]:
from math import *
r = float(input())
c = 2 * pi * r
s = pi * pow(r, 2)
print(s)
print(c)

554.6
966294.7126386268
3484.654571361799


# 28. Средние значения
В математике выделяют следующие средние значения:
* среднее арифметическое чисел a и b - a + b / 2
* среднее геометрическое чисел a и b - sqrt(a * b)
* среднее гармоническое чисел a и b - 2ab / a + b
* среднее квадратичное чисел a и b - sqrt(a^2 + b^2 / 2)

`Формат входных данных`

На вход программе подается два вещественных числа aa и bb​, каждое на отдельной строке.

`Формат выходных данных`

Программа должна вывести 4 числа – среднее арифметическое, геометрическое, гармоническое и квадратичное.

In [24]:
from math import *
a, b = float(input()), float(input())
c = (a + b) / 2
d = sqrt(a * b)
e = (2 * a * b) / (a + b)
f = sqrt((a ** 2 + b ** 2) / 2)
print(c)
print(d)
print(e)
print(f)

35.1
40.3
37.7
37.61023796787252
37.52068965517241
37.78954881974644


# 29. Тригонометрическое выражение
Напишите программу, вычисляющую значение тригонометрического выражения sinx+cosx+tan^2x по заданному числу градусов x.

`Формат входных данных`

На вход программе подается одно вещественное число xx измеряемое в градусах​. 

`Формат выходных данных`

Программа должна вывести одно число – значение тригонометрического выражения.

Примечание 1. Тригонометрические функции принимают аргумент в радианах. Чтобы перевести градусы в радианы, воспользуйтесь формулой r = x * pi / 180

Примечание 2. Модуль math содержит встроенную функцию radians(), которая переводит угол из градусов в угол в радианах.

In [25]:
from math import *
x = radians(float(input()))
a = sin(x) + cos(x) + tan(x)**2
print(a)

30.0
1.6993587371177719


# 30. Пол и потолок
Напишите программу, вычисляющую значение ⌈x⌉ +⌊x⌋ по заданному вещественному числу x.

`Формат входных данных`

На вход программе подается одно вещественное число x.

`Формат выходных данных`

Программа должна вывести одно число – значение указанного выражения.

Примечание. ⌈x⌉ – потолок числа, ⌊x⌋ – пол числа.

In [26]:
from math import *
x = float(input())
y = ceil(x)
z = floor(x)
a = y + z
print(a)

5.5
11


# 31. Квадратное уравнение 🌶️🌶️
Даны три вещественных числа a, b, c. Напишите программу, которая находит вещественные корни квадратного уравнения ax^2 + bc + c = 0. 

`Формат входных данных`

На вход программе подается три вещественных числа a != 0, b, c, каждое на отдельной строке.

`Формат выходных данных`

Программа должна вывести вещественные корни уравнения если они существуют или текст «Нет корней» в противном случае.

Примечание. Если уравнение имеет два корня, то следует вывести их в порядке возрастания.

In [27]:
from math import *
a, b, c = float(input()), float(input()), float(input())
d = b ** 2 - 4 * a * c
if d < 0:
    print("Нет корней")
elif d == 0:
    x = -b / (2 * a)
    print(x)
elif d > 0:
    x1 = (-b + sqrt(d)) / (2 * a)
    x2 = (-b - sqrt(d)) / (2 * a)
    if x1 > x2:
        print(x2)
        print(x1)
    else:
        print(x1)
        print(x2)

1
2
1
-1.0


# 32. Правильный многоугольник
Правильный многоугольник — выпуклый многоугольник, у которого равны все стороны и все углы между смежными сторонами. Площадь правильного многоугольника с длиной стороны aa и количеством сторон nn вычисляется по формуле: 
S = n * a^2 / 4tg(pi/n)

Даны два числа: натуральное число nn и вещественное число aa. Напишите программу, которая находит площадь указанного правильного многоугольника.

`Формат входных данных`

На вход программе подается два числа n и a, каждое на отдельной строке.

`Формат выходных данных`

Программа должна вывести вещественное число – площадь многоугольника.

In [28]:
from math import *
n, a = int(input()), float(input())
s = (n * a ** 2) / (4 * tan(pi / n))
print(s)

3
2.0
1.7320508075688779
